src/lib/add-on/clients/add-on_auto.rb

```diff
@@ -211,6 +211,10 @@ def create_source(add_on, product, media_url)
 
           return :continue
         end
+      ensure
+        # Restore the unexpanded URL to have the original URL
+        # in the saved /etc/zypp/repos.d file (bsc#972046, bsc#1194851).
+        Pkg.SourceChangeUrl(source_id, media_url)
```

Comment:
well, you should at least protect against case when source_id is nil, -1 like at https://github.com/yast/yast-add-on/pull/120/files#diff-047306365196c0d61bdd02db76a557b4ade6b36639b8ef202ceb8f869dcc1aecR193

and what about case when Source is deleted? https://github.com/yast/yast-add-on/pull/120/files#diff-047306365196c0d61bdd02db76a557b4ade6b36639b8ef202ceb8f869dcc1aecR193

Reply:
I've changed when the URL is restored. Please, check it again. Thanks.